### DISCUSSION TITLE
Azure Devops deployments with user secrets

### DISCUSSION THREAD

**Discussion by gary-walker-82** [Azure/azure-dev#1459] [Q&A]
Hi all,
im playing around with the Azure Developer CLI and all the templates use User Secrets,  the cli itself initialises these on the project.  when this happens in Azure Devops the pipeline crashes with a FileNotFound error, the path looks correct and this was taken from the azure.yaml file so i have to have the repo loaded correctly. 
![image](https://user-images.githubusercontent.com/13643656/214709060-fa9ce89f-00cf-443f-9a83-8d311156ceac.png)

ive also tried running the `dotnet user-secret init` command manually in a separate step and get the same result.
Has anyone successfully run the provision step in an Azure DevOps pipeline?  
Is it possible to turn off user secrets?

**Comment by vhvb1989** [comment]
Hello @gary-walker-82 

I'd like to understand a little more about where you are getting this issue. Can you mention:
- What template(s) are you trying ( just want to double check if you might be using a terraform template ).
- What `azd` are you running?  (for example `azd init + azd pipeliene config`  or `azd up + azd pipeline config` , or else)

The azdo pipelines currently depend on the task `AzureCLI@2`, which is used to authenticate to Azure.
When `azd pipeline config` runs, a `service connection` is created within the azdo organization and that's what the `AzureCLI@2` uses.

So, I will be happy to help you here, just need a little more information.
